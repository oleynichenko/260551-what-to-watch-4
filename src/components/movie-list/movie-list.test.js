import MoviesList from './movie-list.jsx';

const moviesList = [
  {
    id: 1,
    title: `Теория большого взрыва: Откровение ринита`,
    image: `path`,
  },
  {
    id: 2,
    title: `Звездный путь`,
    image: `path`,
  },
  {
    id: 3,
    title: `Рик и Морти: Риконечная Мортистория`,
    image: `path`,
  },
];

it(`should render MovieList with 3 movie cards`, () => {
  const onMovieTitleClick = jest.fn();

  const tree = renderer
    .create(
        <MoviesList
          moviesList={moviesList}
          onMovieTitleClick={onMovieTitleClick}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});

