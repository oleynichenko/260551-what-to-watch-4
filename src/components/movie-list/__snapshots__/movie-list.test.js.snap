// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieList with 3 movie cards 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Теория большого взрыва: Откровение ринита"
        height="175"
        src="path"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Теория большого взрыва: Откровение ринита
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Звездный путь"
        height="175"
        src="path"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Звездный путь
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Рик и Морти: Риконечная Мортистория"
        height="175"
        src="path"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Рик и Морти: Риконечная Мортистория
      </a>
    </h3>
  </article>
</div>
`;
